addon_imps/storage/figshare.py

```diff
@@ -0,0 +1,216 @@
+        else:
+            result = []
+        return ItemSampleResult(
+            items=[item.item_result for item in result],
+            next_sample_cursor=str(cursor + 1),
+        )
+
+    async def _fetch_articles(self, page_cursor: int) -> list[ItemResultable]:
+        async with self.network.GET(
+            "account/articles",
+            query={
+                "page": page_cursor,
+                "page_size": 20,
+            },
+        ) as response:
+            return [
+                Article.from_json(project) for project in await response.json_content()
+            ]
+
+    async def _fetch_project_articles(
+        self, project_id: int | str, page_cursor: int
+    ) -> list[Article]:
+        async with self.network.GET(
+            f"account/{project_id}/articles",
+            query={
+                "page": page_cursor,
+                "page_size": 20,
+            },
+        ) as response:
+            json = await response.json_content()
+            return [Article.from_json(project) for project in json]
+
+    async def _fetch_article_files(
+        self, article_id: str, page_cursor: int
+    ) -> list[File]:
+        async with self.network.GET(
+            f"account/{article_id}/files",
+            query={
+                "page": page_cursor,
+                "page_size": 20,
+            },
+        ) as response:
+            json = await response.json_content()
+            return [File.from_json(project) for project in json]
+
+    async def _fetch_projects(self, page: int) -> list[ItemResultable]:
+        async with self.network.GET(
+            "account/projects",
+            query={
+                "page": page,
+                "page_size": 20,
+            },
+        ) as response:
+            json = await response.json_content()
+            return [Project.from_json(json_project) for json_project in json]
+
+    async def _fetch_article(self, article_id: int | str) -> Article:
+        assert self._is_article_id(article_id)
+
+        async with self.network.GET(
+            f"account/{article_id}",
+        ) as response:
+            return Article.from_json(await response.json_content())
+
+    def _is_article_id(self, article_id: str) -> bool:
+        return (
+            "articles" in article_id
+            and (article_id.startswith("projects/") and article_id.count("/") == 3)
+            or (article_id.startswith("articles/") and article_id.count("/") == 1)
+        )
+
+    @staticmethod
+    def _is_project_id(project_id: str) -> bool:
+        return project_id.startswith("projects/") and project_id.count("/") == 1
+
+    async def _fetch_project(self, project_id: str) -> Project:
+        assert project_id.startswith("projects/")
+
+        async with self.network.GET(
+            f"account/{project_id}",
+        ) as response:
+            return Project.from_json(await response.json_content())
+
+
+###
+# module-local helpers
```

Comment:
I wonder if this comment necessary, noticed that you have the same ones later on?

Reply:
The same comment was in different file